pkg/v1/tkg/client/validate.go

```diff
@@ -1837,34 +1839,202 @@ func getDockerBridgeNetworkCidr() (string, error) {
+	if err != nil {
+		return nil, err
+	}
+	aviPassword, err := c.TKGConfigReaderWriter().Get(constants.ConfigVariableAviControllerPassword)
+	if err != nil {
+		return nil, err
+	}
+	aviCAEncoded, err := c.TKGConfigReaderWriter().Get(constants.ConfigVariableAviControllerCA)
+	if err != nil {
+		return nil, err
+	}
+	aviCA, err := base64.StdEncoding.DecodeString(aviCAEncoded)
+	if err != nil {
+		return nil, err
+	}
+	aviClient := avi.New()
+	// validate avi controller account
+	aviControllerParams := &models.AviControllerParams{
+		Username: aviUserName,
+		Password: aviPassword,
+		Host:     aviController,
+		Tenant:   "admin",
+		CAData:   string(aviCA),
+	}
+	authed, err := aviClient.VerifyAccount(aviControllerParams)
+	if err != nil {
+		return nil, err
+	}
+	if !authed {
+		return nil, errors.Errorf("unable to authenticate avi controller due to incorrect credentials")
+	}
+	return aviClient, nil
+}
+
+// ValidateAVIControllerVersion validates AVI controller version format, it shoulde be something like 20.1.7
+func (c *TkgClient) ValidateAviControllerVersion() error {
+	aviVersion, _ := c.TKGConfigReaderWriter().Get(constants.ConfigVariableAviControllerVersion)
+	if aviVersion == "" {
+		return nil
+	}
+	var re = regexp.MustCompile(constants.AviControllerVersionRegex)
+	if !re.MatchString(aviVersion) {
+		return errors.Errorf("incorrect avi controller version format")
+	}
+	return nil
+}
+
+// ValidateAVICloud validates if configured cloud exists or not
+func (c *TkgClient) ValidateAviCloud(aviClient avi.Client) error {
+	aviCloud, err := c.TKGConfigReaderWriter().Get(constants.ConfigVariableAviCloudName)
+	if err != nil {
 		return err
 	}
-	if err := c.validateNetworkSeparation(constants.ConfigVariableAviControlPlaneNetwork, clusterRole); err != nil {
+	if _, err = aviClient.GetCloudByName(aviCloud); err != nil {
 		return err
 	}
-	if err := c.validateNetworkSeparation(constants.ConfigVariableAviControlPlaneNetworkCidr, clusterRole); err != nil {
+	return nil
+}
+
+// ValidateAVIManagementClusterServiceEngineGroup validates if configured service engine group exists or not
+func (c *TkgClient) ValidateAviServiceEngineGroup(aviClient avi.Client) error {
+	aviServiceEngineGroup, err := c.TKGConfigReaderWriter().Get(constants.ConfigVariableAviServiceEngineGroup)
+	if err != nil {
 		return err
 	}
-	if err := c.validateNetworkSeparation(constants.ConfigVariableAviManagementClusterControlPlaneVipNetworkName, clusterRole); err != nil {
+	if _, err = aviClient.GetServiceEngineGroupByName(aviServiceEngineGroup); err != nil {
 		return err
 	}
-	if err := c.validateNetworkSeparation(constants.ConfigVariableAviManagementClusterControlPlaneVipNetworkCidr, clusterRole); err != nil {
+	return nil
+}
+
+// ValidateAVIManagementClusterServiceEngineGroup validates if configured management cluster service engine group exists or not
+func (c *TkgClient) ValidateAviManagementClusterServiceEngineGroup(aviClient avi.Client) error {
```

Comment:
What is the difference between `ConfigVariableAviServiceEngineGroup` and `ConfigVariableAviManagementClusterServiceEngineGroup`?

Reply:
The first one is required and management cluster service engine group is optional